docs/CI-CD/dockerByPR.yml

```diff
@@ -0,0 +1,98 @@
+name: DockerCI-PR
+
+on:
+  pull_request:
+    branches:
+      - master
+
+    paths-ignore:
+      - 'docs/**'
+      - '.github/**'
+      - '**.md'
+      - 'saveenv.sh'
+
+jobs:
+
+  docker-build-push:
+
+    runs-on: ubuntu-latest
+    env:
+      #  DOCKER_REPO  - the docker repository name in the form: organization/repository
+      DOCKER_REPO: 'retaildevcrew/helium-csharp'
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Login to registry
+      run: |
+        # Login to registry
+
+        #    You must add the following secrets to github
+        #    DOCKER_USER  - your dockerhub user name
+        #    DOCKER_PAT   - the personal access token for dockerhub
+       
+
+        ## login to dockerhub
+        echo "${{ secrets.DOCKER_PAT }}" | docker login -u ${{ secrets.DOCKER_USER }} --password-stdin
+
+        ## login to Azure
+        az login --service-principal  -u ${{ secrets.SERVICE_PRINCIPAL }} --tenant ${{ secrets.TENANT }} -p ${{ secrets.SERVICE_PRINCIPAL_SECRET }}
+        
+    - name: Test App
+      run: |
+        # run the app tests
+        cd src/tests
+        dotnet restore > nil
+        dotnet test --nologo /p:CollectCoverage=true
+        cd ../..
+        
+    - name: Build Container
+      run: docker build . --file Dockerfile --tag image
+
+    - name: Docker Push
+      run: |
+        # Tag and push the docker image
+
+        IMAGE_ID=$DOCKER_REPO
+
+        # Tag image based on repo tag if a github label
+
+
+        # Strip git ref prefix from version
+        VERSION=$(echo "${{ github.head_ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+        #docker tag image $IMAGE_ID:dev-$VERSION
+        #docker push $IMAGE_ID:dev-$VERSION
+
+        docker tag image $IMAGE_ID:beta
+        docker push $IMAGE_ID:beta
+
+  acr-build-push-version:
```

Comment:
This should be a separate file. Our customers aren't usually going to use dockerhub and ACR